src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -50,7 +49,8 @@
     * for the actual HTML fragment for the configuration screen.
     */
     @DataBoundConstructor
-    public DockerBuilder(String repoName, String repoTag, boolean skipPush, boolean noCache, boolean forcePull, boolean skipBuild, boolean skipDecorate, boolean skipTagLatest, String dockerfilePath) {
+    public DockerBuilder(DockerRegistryEndpoint registry, String repoName, String repoTag, boolean skipPush, boolean noCache, boolean forcePull, boolean skipBuild, boolean skipDecorate, boolean skipTagLatest, String dockerfilePath) {
```

Comment:
Yes it has grown out of control. Good tip.
On Fri, 1 May 2015 at 7:26 am Jesse Glick <notifications@github.com> wrote:

> In src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java
> <https://github.com/jenkinsci/docker-build-publish-plugin/pull/16#discussion_r29472192>
> :
>
> > @@ -50,7 +49,8 @@
> >      * for the actual HTML fragment for the configuration screen.
> >      */
> >      @DataBoundConstructor
> > -    public DockerBuilder(String repoName, String repoTag, boolean skipPush, boolean noCache, boolean forcePull, boolean skipBuild, boolean skipDecorate, boolean skipTagLatest, String dockerfilePath) {
> > +    public DockerBuilder(DockerRegistryEndpoint registry, String repoName, String repoTag, boolean skipPush, boolean noCache, boolean forcePull, boolean skipBuild, boolean skipDecorate, boolean skipTagLatest, String dockerfilePath) {
>
> I would recommend putting only truly mandatory parameters with no sensible
> default in the @DataBoundConstructor. All else can get @DataBoundSetters.
> That makes it much easier to maintain Java-level compatibility later.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/docker-build-publish-plugin/pull/16/files#r29472192>
> .
>